src/pedalintandem.py

```diff
@@ -0,0 +1,212 @@
+
+
+	dates = []
+	date_opts = offers_selector.select('div.product-variations-variety select[name="variety_id"] option')
+	for date_opt in date_opts:
+		booking_begin = datetime.strptime(date_opt['data-booking-begin-at'], "%Y-%m-%d %H:%M:%S %Z").astimezone(IST).isoformat()
+		startdate = datetime.strptime(date_opt.get_text().strip(), "%d-%b-%Y").astimezone(IST).isoformat()
+		endDate = (datetime.fromisoformat(startdate) + timedelta(hours = duration_in_hours)).isoformat()
+
+		dates.append({"startdate": startdate, "endDate": endDate, "availabilityStarts": booking_begin})
+	
+	# details
+	metrics = {}
+	event_metrics = event.select('div.single-metric.active div.content')
+
+	for event_metric in event_metrics:
+		title = event_metric.find('p').get_text().strip()
+		value = event_metric.find('h3').get_text().strip()
+		metrics[title] = value
+
+	description = event.select_one('div.trix-content div').get_text()
+
+	return {
+		"name": heading,
+		"location": location,
+		"priceCurrency": "INR",
+		"offers": offers,
+		"dates": dates,
+		"duration": duration_in_hours,
+		"description": [
+		process_description(description),
+		metrics
+		]
+	}
+
+def convert_to_24_hour(start_time):
+    # Extract the numeric part of the time
+    time_match = re.search(r'[0-9]+(:[0-9]+)?', start_time)
+    
+    if time_match:
+        time_str = time_match.group(0)
+        # Split hours and minutes
+        if ':' in time_str:
+            hours, minutes = map(int, time_str.split(':'))
+        else:
+            hours = int(time_str)
+            minutes = 0
+
+        # Check for 'am' or 'pm' and adjust hours accordingly
+        if 'am' in start_time.lower():
+            if hours == 12:  # Special case for 12 am
+                hours = 0
+        elif 'pm' in start_time.lower():
+            if hours != 12:  # Special case for 12 pm
+                hours += 12
+
+        # Convert time to decimal format
+        total_hours = hours + minutes / 60.0
+        return total_hours
+
+def convert_duration_in_hours(duration):
+	duration_range = duration.split(',')[0]
+
+	# fetch the upper limit of time duration
+	if bool(re.search("hour", duration)):
+		if bool(re.search('to', duration_range)):
+			duration_in_hours = duration_range.replace("hours", "").split('to')[1].strip()
+		elif bool(re.search('-', duration_range)):
+			duration_in_hours = duration_range.replace("hours", "").split('-')[1].strip()
+		else:
+			duration_in_hours = duration_range.replace("hours", "").strip()
+
+	elif bool(re.search("hrs", duration)):
+		if bool(re.search('to', duration_range)):
+			duration_in_hours = duration_range.replace("hrs", "").split('to')[1].strip()
+		elif bool(re.search('-', duration_range)):
+			duration_in_hours = duration_range.replace("hrs", "").split('-')[1].strip()
+		else:
+			duration_in_hours = duration_range.replace("hrs", "").strip()
+
+	else:
+		duration_in_hours = 0
+
+	return int(duration_in_hours)
+
+def process_description(description):
```

Comment:
Take details dict here, and include it in the description perhaps

```suggestion
def process_description(description, details: dict):
```

Reply:
Concatenated directly